huaweicloud/services/dcs/resource_huaweicloud_dcs_instance.go

```diff
@@ -0,0 +1,1008 @@
+				"creating DCS instance. %s : %#v", d.Id(), err)
+		}
+	}
+
+	// wait for the instance to be created successfully and in running state
+	err = waitForDcsInstanceCompleted(ctx, client, id, d.Timeout(schema.TimeoutCreate),
+		[]string{"CREATING"}, []string{"RUNNING"})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// create whiteList if configured.
+	if d.Get("whitelists").(*schema.Set).Len() > 0 {
+		whitelistOpts := buildWhiteListParams(d)
+		logp.Printf("[DEBUG] Create whitelist options: %#v", whitelistOpts)
+
+		err = whitelists.Put(client, id, whitelistOpts).ExtractErr()
+		if err != nil {
+			return fmtp.DiagErrorf("Error creating whitelist for DCS instance (%s): %s", id, err)
+		}
+		// wait for whitelist created
+		err = waitForWhiteListCompleted(ctx, client, d)
+		if err != nil {
+			return fmtp.DiagErrorf("Error while waiting to create DCS whitelist: %s", err)
+		}
+	}
+
+	return resourceDcsInstancesRead(ctx, d, meta)
+}
+
+func waitForDcsInstanceCompleted(ctx context.Context, c *golangsdk.ServiceClient, id string, timeout time.Duration,
+	padding []string, target []string) error {
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      padding,
+		Target:       target,
+		Refresh:      refreshDcsInstanceState(c, id),
+		Timeout:      timeout,
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmtp.Errorf("[DEBUG] Error while waiting to create/resize/delete DCS instance. %s : %#v",
+			id, err)
+	}
+	return nil
+}
+
+func refreshDcsInstanceState(c *golangsdk.ServiceClient, id string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		r, err := instances.Get(c, id)
+		if err != nil {
+			err404 := golangsdk.ErrDefault404{}
+			if errors.As(err, &err404) {
+				return &(instances.DcsInstance{}), "DELETED", nil
+			}
+			return nil, "Error", err
+		}
+		return r, r.Status, nil
+	}
+}
+
+func resourceDcsInstancesRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	client, err := conf.DcsV2Client(conf.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("error creating HuaweiCloud DCS Client(v2): %s", err)
+	}
+
+	r, err := instances.Get(client, d.Id())
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "DCS instance")
+	}
+	logp.Printf("[DEBUG] Get DCS instance : %#v", r)
+
+	// capacity
+	capacity := r.Capacity
+	if capacity == 0 {
+		capacity, _ = strconv.ParseFloat(r.CapacityMinor, floatBitSize)
+	}
+	azCodes, err := getAvailableZoneIDByCode(client, r.AzCodes)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// batch set attributes
+	mErr := multierror.Append(nil,
+		d.Set("region", conf.GetRegion(d)),
+		d.Set("name", r.Name),
+		d.Set("engine", r.Engine),
+		d.Set("engine_version", r.EngineVersion),
+		d.Set("capacity", capacity),
+		d.Set("flavor", r.SpecCode),
+		d.Set("available_zones", azCodes),
```

Comment:
I think it's unnecessary to set available_zones field as it was deprecated.

Reply:
done

Reply:
done